### ISSUE TITLE
can not be used in todo app in real time 

### ISSUE THREAD

**Issue by neerajSharma001** [blackhorse-one/stomp_dart_client#16]
hi , 
i am building a todo app for like chat system ... in which multiple users can add/update todoList .

i am able to get broadcast msg when any user add ... but can not update todoList. 

i did not understand how i pass todoList variable or any other variable in callback .

**Comment by KammererTob** [comment]
Hey,

would you mind posting some code (minimal working example)? It is hard to follow what your actual problem is, without having it in code.

Thank you

**Comment by KammererTob** [comment]
I am going to close this issue because of inactivity. If you feel like your issue is not yet solved, then please feel free to open it.

**Comment by neerajSharma001** [comment]
Hi ,

`dynamic onConnect(StompClient client, StompFrame frame) {
  print("Stomp Connected ...");

  client.subscribe(
      destination: '/topic/greetings',
      // headers: {},
      callback: (StompFrame frame) {
        // Received a frame for this subscription
        print("inside subscriber");
     
    var todoState = Provider.of<TodoState>(context); -- unable to use provider 
        ToDoService.handleBroadCastResponse(todoState,frame.body);
        // todoState.addTaskInList(1);
      });


}`


In short i want to add an item in list ... but unable to do .
currently i am using ChangeNotifierProvider .

**Comment by KammererTob** [comment]
Hi,

do you have this code inside initState? Or what is the error you are getting? It is probably because flutter cannot find the Provider inside your current context. I probably would need more information (code/errors etc.) to help you out.

**Comment by neerajSharma001** [comment]
Hi, Thanks for your response . 
you are right , i am trying to make stompClient as final variable that's why it was unable to use context. 
now i am using it as a non-final variable. and it is working fine.
